### DISCUSSION TITLE
network permission + local ics file

### DISCUSSION THREAD

**Discussion by 88-percent** [bitfireAT/icsx5#385] [Q&A] [answered]
I subscribe to a single local ics file.  It seems that automatic syncing doesn't work unless I grant ICSx5 network permissions. 

Is this the expected behavior?

Manual sync works fine without the permission - it appears that only the automatic sync isn't triggered.  It doesn't appear that ICSx5 is polling a server for the time as automatic sync works with airplane mode enabled.

I've read the [privacy policy](https://icsx5.bitfire.at/privacy/) and I'm really impressed with it, but I'd still prefer not to grant unneeded permissions. I'm particularly wary of the network permission. 

If this isn't just a glitch on my end owing to my setup is there any technical reason for ICSx5 not to sync local files in the background without the network permission?

**Comment by sunkup** [comment, marked as answer]
> Is this the expected behavior?

No. Well, let's say we did not expect this use case. Usually people do still allow for network access.

> If this isn't just a glitch on my end owing to my setup is there any technical reason for ICSx5 not to sync local files in the background without the network permission?

No, not really. At the moment we use Androids work manager constraints, to require network connectivity before a [manual](https://github.com/bitfireAT/icsx5/blob/dev/app/src/main/java/at/bitfire/icsdroid/SyncWorker.kt#L60) or [periodic](https://github.com/bitfireAT/icsx5/blob/dev/app/src/main/java/at/bitfire/icsdroid/PeriodicSyncWorker.kt#L30) sync runs. With some rewriting, it should be possible to run the sync without the network requirement / network permission. 

We are kind of busy with DAVx5 at the moment. If you know android and kotlin: would you be able to provide a PR?

**Reply by ArnyminerZ** [reply]
Followup https://github.com/bitfireAT/icsx5/issues/388

**Reply by 88-percent** [reply]
> We are kind of busy with DAVx5 at the moment. If you know android and kotlin: would you be able to provide a PR?

Understood. DAVx5 should definitely be the priority. Sorry I can't help  with a PR.
